houston/ardu/command_factory.py

```diff
@@ -0,0 +1,95 @@
+import os
+import logging
+import yaml
+from typing import Dict, Any, Type, List
+
+from .connection import CommandLong
+from ..valueRange import ContinuousValueRange, DiscreteValueRange
+from ..command import Parameter, Command, CommandMeta
+from ..specification import Idle
+
+logger = logging.getLogger(__name__)  # type: logging.Logger
+logger.setLevel(logging.DEBUG)
+
+
+def create_command(command: Dict[str, Any]) -> Type[Command]:
+    """
+    From a given dictionary, generates the Command class.
+    """
+    try:
+        name = command['name']
+    except KeyError:
+        msg = "missing 'name' field of Command"
+        raise TypeError(msg)
+    try:
+        id = command['id']
+    except KeyError:
+        msg = "missing 'id' field of Command"
+        raise TypeError(msg)
+    parameters = []
+    params_name = {}
+    for i in range(1, 8):
+        p = 'p{}'.format(i)
+        if p in command:
+            param = None
+            try:
+                p_name = command[p]['name']
+            except KeyError:
+                msg = "missing 'name' field of Command parameter {}".format(p)
+                raise TypeError(msg)
+            try:
+                typ = command[p]['value']['type']
+            except KeyError:
+                msg = "missing 'value' or 'type' field of Command parameter {}"
+                msg = msg.format(p)
+                raise TypeError(msg)
+            if typ == 'discrete':
+                vals = command[p]['value']['vals']
+                param = Parameter(p_name, DiscreteValueRange(vals))
+            elif typ == 'continous':
+                min_value = command[p]['value']['min']
+                max_value = command[p]['value']['max']
+                param = Parameter(p_name, ContinuousValueRange(min_value,
+                                                               max_value,
+                                                               True))
+            else:
+                msg = "The type of value {} is not supported".format(typ)
+                raise Exception(msg)
+            parameters.append(param)
+            params_name['param_{}'.format(i)] = p_name
+        else:
+            params_name['param_{}'.format(i)] = 0
+
+    def to_message(self):
+        params = {}
+        for p, n in params_name.items():
```

Comment:
`params`, `params_names` and `parameters` all seem very similar. It's not entirely clear what's going on here?